frontend/src/components/modals/RateTaskModal.tsx

```diff
@@ -138,28 +140,20 @@ export const RateTaskModal: Modal<ModalTypes.RATE_TASK_MODAL> = ({
         !rating.customer || rating.customer.roadmapId === task?.roadmapId,
     )
     // show only new or existing ratings depending on edit value
-    .filter(({ value }) => (edit ? value : !value));
+    .filter(({ value }) => (edit ? value : !value))
+    .map((rating) => ({
+      ...rating,
+      ...(!rating.comment && { comment: undefined }),
+    }));
```

Comment:
When editing a rating this will make a distinction between untouched rating with no comment (comment is `undefined`) and a rating where empty comment is first edited but then cleared (comment is the empty string).

This shows in the submit button being enabled in the latter case, even though no meaningful change has been made.

So, unless I'm missing something, I think this should be removed.

Reply:
I think it related to the logic in detecting changed edited ratings. Improved the predicate so it is not needed anymore.